controllers/datamovement_controller.go

```diff
@@ -214,40 +254,154 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	go func() {
 		var cmd *exec.Cmd
 
-		// TODO: UserId/GroupId
-
-		if isTestEnv() {
-			cmd = exec.CommandContext(ctxCancel, "sleep", "1")
+		// Allow the data movement command to be overridden via the configmapa
+		// for testing purposes
+		if dmCmd != "" {
+			cmdList := strings.Split(dmCmd, " ")
+			cmd = exec.CommandContext(ctxCancel, cmdList[0], cmdList[1:]...)
+			// Otherwise use the actual `mpirun dcp` command
 		} else {
+			// TODO: UserId/GroupId
 			cmd = exec.CommandContext(ctxCancel,
 				"mpirun",
 				"--allow-run-as-root",
-				"-np", fmt.Sprintf("%d", len(hosts)), // # TODO: Might want to adjust this if running on a rabbit node
+				"-np", fmt.Sprintf("%d", numProcesses), // # TODO: Might want to adjust this if running on a rabbit node
 				"--host", strings.Join(hosts, ","),
-				"dcp", dm.Spec.Source.Path, dm.Spec.Destination.Path)
+				// Dump progress every second to ensure progress collection has
+				// something to collect even if the interval is not 1 second.
+				"dcp", "--progress", "1", dm.Spec.Source.Path, dm.Spec.Destination.Path,
+			)
 		}
-		log.Info("Running Command", "cmd", cmd.String())
+		cmdStatus.Command = cmd.String()
```

Comment:
I think cmd.Exec() can be initialized before the go routine, and you'll get the cmdStatus.Command value to be updated in the Status().Update() on line 240.